fido-uaf-core/src/main/java/org/ebayopensource/fido/uaf/ops/RegistrationResponseProcessing.java

```diff
@@ -158,12 +161,9 @@ private String getAuthenticatorVersion(Tags tags) {
             + tags.getTags().get(TagsEnum.TAG_ASSERTION_INFO.id).value[1];
     }
 
-    private void checkAssertions(RegistrationResponse response)
-        throws Exception {
-        if (response.getAssertions() != null && response.getAssertions().length > 0) {
-            return;
-        } else {
-            throw new Exception("Missing assertions in registration response");
+    private void checkAssertions(RegistrationResponse response) throws AssertionException {
+        if (response.getAssertions() == null && response.getAssertions().length <= 0) {
```

Comment:
Won't `response.getAssertions().length` throw a NPE if the first condition is true? Should probably be `||` instead of `&&`